### PR TITLE
Increased TPS averages number and improved tps command

### PR DESCRIPTION
**Pull request by LB45440078L**
Ok I'll explain briefly what I've done, and why I have done it:
I've improved and added a new argument to the default Spigot /tps command, which didn't look good in my opinion. First the command would take any argument and still display the tps values, which in my opinion is not right as there should be no arguments in order for the command to be executed succesfully. Second I've added the option for the user to add 'graph' as first argument after their /tps command. This will send the user an improved version of the default TPS with a graph.

In order to achieve these I had to create add more TPS averages because the graph could not be made only from 3 values. I've added a map of RollingAverage in the MinecraftServer , where each has as key its correspondent minute.

## The graph:

The graph is generated this way:
the **ordinate** represents the time of the graph, which obviously goes from left to right, and on the **abscissa** we will have the TPS average value.
In order to not flood the chat , I've made 2TPS as the minimum unit on the abscissa, so the output message will be 10 rows tall.
A sample of the ouput would be this:
tps graph

In this photo you can see that in the last minutes the tps had dropped below 19, so the last column is 1 unit lower.

---
I hope you will like my commit and accept it.

-- **TheViperShow**

### CODE REVIEW

File: Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+ 
+@@ -546,11 +553,11 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
+         File mcMeta = new File(bukkitDataPackFolder, "pack.mcmeta");
+         try {
+             com.google.common.io.Files.write("{\n"
+-                    + "    \"pack\": {\n"
+-                    + "        \"description\": \"Data pack for resources provided by Bukkit plugins\",\n"
+-                    + "        \"pack_format\": " + SharedConstants.getGameVersion().getPackVersion() + "\n"
+-                    + "    }\n"
+-                    + "}\n", mcMeta, com.google.common.base.Charsets.UTF_8);
```
Comment (kashike) [inline Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch]: do not modify this section, you have no reason to do so

File: Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
++                        recentTps[1] = rollingAverageMap.get(5).getAverage();
++                        recentTps[2] = rollingAverageMap.get(15).getAverage();
+                         // Paper end
+                         tickSection = curTime;
+                     }
+@@ -1125,26 +1132,26 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
+         }
+ 
+         //if (autosavePeriod > 0 && this.ticks % autosavePeriod == 0) { // CraftBukkit // Paper - move down
+-            //MinecraftServer.LOGGER.debug("Autosave started"); // Paper
```
Comment (kashike) [inline Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch]: do not modify this section, you have no reason to do so

File: Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index b9a398bc..f9585af8 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -1414,7 +1414,7 @@ public final class CraftServer implements Server {
+         GameProfile profile;
+         // Only fetch an online UUID in online mode
+         if (net.minecraft.server.MinecraftServer.getServer().getOnlineMode()
+-                || (org.spigotmc.SpigotConfig.bungee && com.destroystokyo.paper.PaperConfig.bungeeOnlineMode)) {
++            || (org.spigotmc.SpigotConfig.bungee && com.destroystokyo.paper.PaperConfig.bungeeOnlineMode)) {
```
Comment (kashike) [inline Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch]: do not modify this section, you have no reason to do so

File: Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+             profile = console.getUserCache().getProfile( name );
+         } else {
+             // Make an OfflinePlayer using an offline mode UUID since the name has no profile
+@@ -1436,7 +1436,7 @@ public final class CraftServer implements Server {
+             GameProfile profile = null;
+             // Only fetch an online UUID in online mode
+             if ( getOnlineMode()
+-                    || com.destroystokyo.paper.PaperConfig.isProxyOnlineMode() ) // Paper - Handle via setting
++                || com.destroystokyo.paper.PaperConfig.isProxyOnlineMode() ) // Paper - Handle via setting
```
Comment (kashike) [inline Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch]: do not modify this section, you have no reason to do so

File: Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+             {
+                 profile = console.getUserCache().getProfile( name );
+             }
+@@ -1515,11 +1515,11 @@ public final class CraftServer implements Server {
+         Validate.notNull(type, "Type cannot be null");
+ 
+         switch(type){
+-        case IP:
```
Comment (kashike) [inline Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch]: do not modify this section, you have no reason to do so

File: Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
++    }
++
+     // Paper end
+ 
+     private final Spigot spigot = new Spigot()
+@@ -2158,8 +2179,8 @@ public final class CraftServer implements Server {
+     @Override
+     public boolean reloadCommandAliases() {
+         Set<String> removals = getCommandAliases().keySet().stream()
+-                .map(key -> key.toLowerCase(java.util.Locale.ENGLISH))
```
Comment (kashike) [inline Spigot-Server-Patches/0450-Increased-TPS-averages-number-and-improved-tps-comma.patch]: do not modify this section, you have no reason to do so

**Review by kashike — CHANGES_REQUESTED** [review]
etc

**Comment by LB45440078L** [comment]
I'll try to fix this . . .

**Comment by kickash32** [comment]
You don't need to close a PR to fix issues. Just commit changes to your PR branch and the PR will be updated automagically.

**Pull request closed without merging**